src/async_impl/client.rs

```diff
@@ -1098,6 +1098,31 @@ impl Client {
     }
 }
 
+#[cfg(feature = "cookies")]
+impl Client {
+    /// Returns a reader lock on the `CookieStore` (immutable access).
+    /// Requests will be blocked until the lock is dropped.
+    pub fn get_cookies(
+        &self
+    ) -> Option<std::sync::RwLockReadGuard<'_, cookie::CookieStore>> {
```

Comment:
I'd probably rather not reveal that an `RwLock` is used internally, so that way we can change that later and it won't break this API. So, perhaps getting the cookie store from a client should just clone a snapshot. What do you think?

Reply:
Yes, it's not the best solution. But I dislike the idea of returning a cloned snapshot. Currently, the type of cookie_store is `RwLock<cookie::CookieStore>`. But what if we remove the lock and put it inside the CookieStore? The lock would be hidden and used by internal methods. This way, I think we could return a pretty reference to the CookieStore.